.gitlab-ci.yml

```diff
@@ -1,123 +1,98 @@
 stages:
-  - build_base_images
-  - build_derived_images
-
-.build: &template_build
-  tags: 
-    - docker-image-build
-  except: 
-    - master
-  script:
-    - echo
+  - build-base-images
+  - build-derived-images
 
-.build_master: &template_build_master
-  tags: 
-    - docker-image-build
-  only: 
-    - master
+# Build container image using Kaniko as recommended in
+#
+#   https://gitlab.cern.ch/gitlabci-examples/build_docker_image
+#
+.build-image:
+  image:
+    name: gitlab-registry.cern.ch/ci-tools/docker-image-builder
+    entrypoint: [""]
   script:
-    - echo
+    - echo "{\"auths\":{\"$CI_REGISTRY\":{\"username\":\"$CI_REGISTRY_USER\",\"password\":\"$CI_REGISTRY_PASSWORD\"}}}" > /kaniko/.docker/config.json
+    - /kaniko/executor --context dir://${CI_PROJECT_DIR}/${CONTEXT} --destination ${CI_REGISTRY_IMAGE}/${CONTEXT}:${CI_COMMIT_REF_SLUG} --build-arg=REGISTRY=${CI_REGISTRY_IMAGE} --build-arg=TAG=${CI_COMMIT_REF_SLUG}
```

Comment:
Does this also only tag `latest` on master? That's important!

Reply:
I think it tags `master` on `master` and I would switch the Acts builds to use the `master` tag.

Reply:
That's not a good idea! `latest` is chosen automatically whenever docker is invoked, so only master should write to `latest`.

Reply:
As discussed: `latest` is not set automatically and will use explicit branch tags e.g. `master` in the main repo. This will be left as-is.